### DISCUSSION TITLE
Which directory should I put an sqlite3 database?

### DISCUSSION THREAD

**Discussion by Ivo6** [beeware/toga#1776] [Q&A] [answered]
I would like to use an Sqlite3 database which already contains tables and data from another project. My goal is to create an Android app which uses the database, and I'm following the Beeware tutorials as a guide. Which directory should I copy the database into? 

Related question: In the app.py, should I use the full directory path to the database, or a relative path?

**Comment by freakboy3742** [comment, marked as answer]
Any folder inside your app package will be accessible at runtime, so that's as good a place as any to put content like SQLite databases. 

You definitley need to use an absolute path for any reference to that database file - A toga app can't guarantee what the current working directory will be at runtime, as it will vary between platforms (and in the case of mobile deployments, will depend on the specific installation), so relative paths will be ambiguous at best.

To assist with this, the Toga app instance has a `paths` attribute that contains a useful path roots that you can work off. For example, in your `startup()` method, you can reference `self.paths.app` - that will be a Pathlib object providing an absolute path to the folder that contains your app.py. You can use this as a base on which to build a location (for example - `self.paths.app / "resources" / "my.db"` would be a platform independent absolute path of a database file in the resources folder for your app (assuming you've got `src/myapp/resources/my.db` in your project).
